hal/network/lwip/esp32/esp32ncpnetif.cpp

```diff
@@ -204,28 +223,21 @@ void Esp32NcpNetif::loop(void* arg) {
 
 int Esp32NcpNetif::up() {
     NetifEvent ev = NetifEvent::Up;
-    int r = os_queue_put(queue_, &ev, 0, nullptr);
-    if (r) {
-        // FIXME: we'll miss this event, just as an attempt to potentially recover
-        // set internal state.
-        up_ = true;
-    }
-    return r;
+    return os_queue_put(queue_, &ev, 0, nullptr);
 }
 
 int Esp32NcpNetif::down() {
     const auto client = wifiMan_->ncpClient();
-    if (client->connectionState() != NcpConnectionState::CONNECTED) {
+    if (client->connectionState() != NcpConnectionState::CONNECTED && client->ncpState() == NcpState::ON) {
```

Comment:
The idea behind calling disable is to force abort the ongoing (blocking) operation. It may take quite a bit of time for NCP state go to `ON`. I'm not sure it's that good of an idea to add this check. We might be performing some initialization currently which we want to abort.

Reply:
I think calling `disbale()` in this `down()` just intends to abort the on-going attempt establishing a connection. If the NCP is transiting to the `ON` state, it's beyond the intention of calling the `disable()` here. On the other side, the below logic:
```
void fn() {
    Network.connect();
    Network.disconnect();
}
```
expects the NCP state to be `ON` and ready to accept AT commands. With the current implementation, the NCP will reside in the `OFF` state if transiting to `ON` state is aborted.

Reply:
It's not guaranteed that the next operation isn't going to be `Network.off()` and I'd argue that's actually a more likely scenario if we take a look at how https://github.com/particle-iot/device-os/blob/develop/system/src/system_network_manager_api.cpp#L277 is implemented for example.

But I agree, it's probably fine to leave things as-is. Aborting an ongoing power-on/initialization attempt potentially prevents us dropping into a connection attempt if we take a look at https://github.com/particle-iot/device-os/pull/2217/files#diff-333366c726fa5a73345b7c9ced8858345aa5db8b5d22add908e50f7045d83c8eR202, but perhaps it's going to take longer for things to settle, especially if we consider your case where we anyway have to settle on a powered state. Should we perhaps do something in `powerDown()` to abort power-on/initialization sequence to speed things up?

Anyway, no strong preferences, but would be nice to make some comparisons with before/after.

Reply:
I'm going to do some tests to compare the time consumption between 3.0 and pre-3.0.